compiler/main.bal

```diff
@@ -46,13 +47,19 @@ public function main(string[] filenames, *Options opts) returns error? {
         gcName: check nback:validGcName(opts.gc),
         debugLevel: check nback:validDebugLevel(opts.debugLevel)
     };
-    int errorFileCount = 0;
+    err:Diagnostic[] diagnosticErr = [];
     foreach string filename in filenames {
         var [basename, ext] = basenameExtension(filename);
         if ext == SOURCE_EXTENSION {
-            CompileError? err = printDiagnostic(compileBalFile(filename, basename, check chooseOutputBasename(basename, opts.outDir), nbackOptions, opts));
+            CompileError? err;
```

Comment:
Just have an boolean option to use HTML format for errors. Define a DiagnosticPrinter object type. Include this in the `comm.diagnostic` module. Then have two classes implementing this, one for text and one for HTML. Have a `flush` method that is called at the end. Put the HTML class in a separate file.

Reply:
fixed